Scripts/Generator/MapLoader.cs

```diff
@@ -0,0 +1,162 @@
+using Godot;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+public class MapLoader : Spatial
+{
+    [Export]
+    public int UnitSize
+    {
+        get
+        {
+            return unitSize;
+        }
+        set
+        {
+            unitSize = value;
+        }
+    }
+
+    [Export(PropertyHint.File)]
+    public String MapPath
+    {
+        get
+        {
+            return mapPath;
+        }
+        set
+        {
+            mapPath = value;
+        }
+    }
+
+    [Export]
+    public Material WallMaterial
+    {
+        get
+        {
+            return wallMaterial;
+        }
+        set
+        {
+            wallMaterial = value;
+        }
+    }
+
+    [Export]
+    public Material FloorMaterial
+    {
+        get
+        {
+            return floorMaterial;
+        }
+        set
+        {
+            floorMaterial = value;
+        }
+    }
+
+    private class Bounds
+    {
+        public int left, right, top, bottom;
+    }
+
+    private int unitSize;
+    private Material wallMaterial;
+    private Material floorMaterial;
+    private String mapPath;
+    private File mapFile;
+
+    private List<int> pixels = new List<int>();
+    private int size;
+    private List<Bounds> bounds = new List<Bounds>();
```

Comment:
May be better to do the methods in a functional way? i.e. remove `pixels`, `bounds`. See other comments below